feat(cms): add CallBaodaozaiIntro model and deprecate CallBaodaozai

ericsen-tsai opened the pull request:
## Summary

This PR introduces a new `CallBaodaozaiIntro` model to replace the existing `CallBaodaozai` model with a more flexible page-based approach. The old model is marked as deprecated and hidden from the UI while maintaining backward compatibility.

## Changes

- **New CallBaodaozaiIntro model**: Added a new list with page-based content management supporting multiple page types (home, article, topic, news, search, lessons, podcasts, aboutUs)
- **Deprecated CallBaodaozai model**: Marked the existing model as deprecated with description and hidden from UI while keeping it functional
- **Database schema updates**: Added new GraphQL schema and Prisma model definitions for CallBaodaozaiIntro
- **Migration support**: Updated schema files to include the new model and its relationships
- **Access control**: Implemented proper role-based access control for the new model (Owner, Admin, Editor roles)
- **UI configuration**: Set up list view with appropriate initial columns and field visibility

## Additional Notes

The new model uses a more flexible approach where content is organized by page type rather than having separate fields for each page. This allows for easier content management and future extensibility. The old model remains functional to ensure no breaking changes for existing implementations.

## Screenshot(s)

<img width="1255" height="712" alt="image" src="https://github.com/user-attachments/assets/3341a2a9-ab07-4b2e-91c0-449ec90e00a9" />


## Reference(s)
- [Notion Page](https://www.notion.so/twreporter/cms-callbaodaozai-2948dbdca93280b88f11caab66b31f51?v=1588dbdca93281dfa43c000c8415265e&source=copy_link)

ericsen-tsai committed:
feat(cms): adjust call baodaozai list and schema

ericsen-tsai committed:
chore(cms): add migration/20251028090217_update_call_baodaozai/migration.sql

nickhsine commented on packages/cms/lists/call-baodaozai.ts:
```diff
@@ -1,68 +1,63 @@
+    aboutUs: text({
+      label: '關於我們',
+      validation: { isRequired: true },
+    }),
```
thought:
是不是要和 PM 確認一下這些欄位是否都是必填欄位？

ericsen-tsai replied:
Yes, they are all required

nickhsine reviewed (approved):
suggestion:
因為 ndx branch 和 prod branch 共用 database，這邊的 migration 要 backward compatible 才不會影響到 prod CMS 的運作。
看是不是要用新增欄位的方式處理，把舊的欄位設為 hidden。


thought:
目前的做法是「一筆資料，不同欄位對上不同頁面」，這個做法會有以下考量：
- 新增一個頁面，就會需要新增一個欄位；新增欄位就會需要 db migration
- 而且欄位會隨著頁面變多而膨脹
- 在 query 時，會需要 query list 回去，再挑選第一筆（有點不大直覺）

不曉得改成「多筆資料，一筆資料對上一個頁面」的做法會不會比較直覺，同時可以解決上述問題？
例如：
```
  fields: {
    page: select({
      type: 'enum',
      options: [
        { label: '首頁', value: 'home' },
        { label: '文章頁', value: 'article' },
        { label: '專題頁', value: 'topic' },
        { label: '新聞集合頁', value: 'news' },
        { label: '搜尋頁', value: 'search' },
        { label: '教案集合頁', value: 'lessons'}
      ],
      validation: { isRequired: true },
      isIndexed: 'unique',
     }),
     content: text(),
```
query 時，可以用 pageUniqueWhereInput 去指定某一個頁面的資料。

ericsen-tsai committed:
Revert "feat(cms): adjust call baodaozai list and schema"

This reverts commit 1ce474a666de9f68196004b9201c6b6ef2441514.

ericsen-tsai committed:
Revert "chore(cms): add migration/20251028090217_update_call_baodaozai/migration.sql"

This reverts commit 39755c8759735424572009ff28656940d667f960.

ericsen-tsai committed:
feat(cms): add CallBaodaozaiIntro model and related GraphQL schema

- Introduced CallBaodaozaiIntro model in Prisma schema with fields for page type, content, and timestamps.
- Added GraphQL types, inputs, and mutations for managing CallBaodaozaiIntro entries.
- Created a new list for CallBaodaozaiIntro with access control and UI configuration.
- Updated index to include the new CallBaodaozaiIntro list.
- Hid the deprecated CallBaodaozai list.

ericsen-tsai commented:
> suggestion: 因為 ndx branch 和 prod branch 共用 database，這邊的 migration 要 backward compatible 才不會影響到 prod CMS 的運作。 看是不是要用新增欄位的方式處理，把舊的欄位設為 hidden。
> 
> thought: 目前的做法是「一筆資料，不同欄位對上不同頁面」，這個做法會有以下考量：
> 
> * 新增一個頁面，就會需要新增一個欄位；新增欄位就會需要 db migration
> * 而且欄位會隨著頁面變多而膨脹
> * 在 query 時，會需要 query list 回去，再挑選第一筆（有點不大直覺）
> 
> 不曉得改成「多筆資料，一筆資料對上一個頁面」的做法會不會比較直覺，同時可以解決上述問題？ 例如：
> 
> ```
>   fields: {
>     page: select({
>       type: 'enum',
>       options: [
>         { label: '首頁', value: 'home' },
>         { label: '文章頁', value: 'article' },
>         { label: '專題頁', value: 'topic' },
>         { label: '新聞集合頁', value: 'news' },
>         { label: '搜尋頁', value: 'search' },
>         { label: '教案集合頁', value: 'lessons'}
>       ],
>       validation: { isRequired: true },
>       isIndexed: 'unique',
>      }),
>      content: text(),
> ```
> 
> query 時，可以用 pageUniqueWhereInput 去指定某一個頁面的資料。

@nickhsine Based on your comment, I already created another schema for `CallBaodaozaiIntro` to prevent confusion and also hid the deprecated `CallBaodaozai` schema. Thanks!

[feat(cms): add CallBaodaozaiIntro model and related GraphQL schema](https://github.com/kids-reporter/kids-reporter-monorepo/pull/804/commits/46eb0e1063eb4f2e707040536d86cad79a2ef2de)

nickhsine commented:
LGTM 👍 👍 👍 
再 merge 前記得要新增 migration script 喔

ericsen-tsai committed:
chore(cms): add migrations/20251029072351_add_call_baodaozai_intro
